### DISCUSSION TITLE
Unable to insert without providing the id which I have set to default UUID generation

### DISCUSSION THREAD

**Discussion by bala-techguy** [supabase/supabase#3174] [Questions] [answered]
I'm very new to Supabase and have been getting my hands on this for the past few days. I'm stuck in the part where I'm trying to insert values to a table without giving an id value which is the primary key. 

![image](https://user-images.githubusercontent.com/23168216/132995815-443b36cb-89f5-4647-8d33-c041c043ae36.png)

The above is the error message I'm getting. I don't see anything wrong with the row-level security that I have defined. 

![image](https://user-images.githubusercontent.com/23168216/132995875-34b195ed-d42c-485a-9e91-ef5857bd96ee.png)

The schema of the table is as follows:

![image](https://user-images.githubusercontent.com/23168216/132996083-9eb18a83-d67a-4e9b-9c2d-2f2df23efc72.png)

Below is my code. 

```
        const user_profiles_data = await supabase
          .from('user_profiles')
          .insert([
            {
              id: user.id, name: payload.fullName, bio: '', photourl: '',
            },
          ]);
        const user_providers_data = await supabase
          .from('user_providers')
          .insert([
            {
              user_profiles_id: user_profiles_data.data[0].id, provider: 'email', email: '',
            },
          ]);

```
I'm inserting into two tables. If you look above, I first insert the values to user_profiles and use that id as one of the fields in my next table. The first table insertion is fine, the error is only on the second table insertion. 

I tried setting an id similar to below and if I do that, it works fine. But I want the id (primary key) to be generated automatically with a default UUID. 

```
        const user_providers_data = await supabase
          .from('user_providers')
          .insert([
            {
              id: '824034ae-6a10-431f-a20a-314a71969729', user_profiles_id: user_profiles_data.data[0].id, provider: 'email', email: '',
            },
          ]);

```

Can someone please help me with this? Already wasted more than 3 hours trying to crack this by myself.

**Comment by steve-chavez** [comment, marked as answer]
**EDIT**: As mentioned below, the problem here was RLS.

@bala41288 Can you try doing the second insertion with an object instead of an array? Like:

```js
const user_providers_data = await supabase
  .from('user_providers')
  .insert(
    {
      user_profiles_id: user_profiles_data.data[0].id, provider: 'email', email: '',
    }
  );
```

**Reply by bala-techguy** [reply]
Thanks for your reply. Just did it, but no luck. :(

![image](https://user-images.githubusercontent.com/23168216/132997682-9b315318-6cfa-4280-bf7e-fe3903aef077.png)

**Reply by steve-chavez** [reply]
Ah, I thought this was a problem regarding default value generation. But seems is just an RLS issue.

Your INSERT policy says `id = uid()`(`id = auth.uid()`), so you'll never be able to generate another uuid for a row since it will always have to match the current user uuid(auth.uid = current user id). Remove that POLICY and this insert should work.

Also have a look at returnning=minimal https://supabase.io/docs/reference/javascript/insert

**Reply by bala-techguy** [reply]
Thank you so much. Yeah RLS was the problem. Changing the RLS pointing to the foreign key instead of the primary key solved the problem for me.
